### DISCUSSION TITLE
2.1.3自动分区如何配置保留数据

### DISCUSSION THREAD

**Discussion by zhangxiangyang** [apache/doris#35052] [A - General / Q&A]
create table if not exists bdp_ods.test
(
    dd_date date not null comment '日期'
)
DUPLICATE KEY(dd_date)
comment 'test'
AUTO PARTITION BY RANGE (date_trunc(dd_date, 'day'))()
DISTRIBUTED BY HASH(dd_date) BUCKETS 1
properties
(
    "dynamic_partition.enable" = "true",
    "dynamic_partition.time_unit" = "day",
    "dynamic_partition.start" = "-60",
    "dynamic_partition.end" = "3",
    "dynamic_partition.prefix" = "p",
    "dynamic_partition.buckets" = "1",
    "replication_num" = "3"
);

以上为建表语句，执行报错：
SQL 错误 [1105] [HY000]: errCode = 2, detailMessage = errCode = 2, detailMessage = Can't use Dynamic Partition and Auto Partition at the same time。
去掉动态分区配置可以成功建表，但是无法指定数据保留天数，2.1.3自动分区新功能能配置数据保留天数吗？

**Comment by ixzc** [comment]
Partitions created by automatic partition will not be deleted, but there is a maximum limit to the size you can specify by setting max_auto_partition_num in fe.conf

**Reply by zhangxiangyang** [reply]
Thank you!
If the max_auto_partition_num limit is exceeded, will Doris delete the partition or will it not allow the partition to be created?

**Reply by ixzc** [reply]
it will not allow to create partitions

**Comment by zclllyybb** [comment]
这个得看你的具体需求是什么？自动的分区生命周期管理是Dynamic partition的功能。2.1.3开始我们禁止了auto partition和Dynamic partition同时使用，因为这个行为结果不好理解。

**Reply by zclllyybb** [reply]
具体而言，你在使用dynamic partition的基础上，还需要auto partition创建分区的原因是什么呢？

**Comment by shawn-happy** [comment]
![image](https://github.com/user-attachments/assets/9a16d0b4-5db4-42c3-8cc6-d3a670f060cb)
我直接执行官网里的sql，也是报
SQL 错误 [1105] [HY000]: errCode = 2, detailMessage = errCode = 2, detailMessage = Can't use Dynamic Partition and Auto Partition at the same time。

**Reply by OnePieceJoker** [reply]
doris版本是多少，2.1.2-2.1.6版本不允许一起用了。

**Reply by zclllyybb** [reply]
2.1后续版本重新支持了这个功能，请更新到最新版。

**Reply by ixzc** [reply]
升级到2.1.7或者3.0.3及以后版本就可以了
